lib/si-settings/src/lib.rs

```diff
@@ -50,4 +49,32 @@ pub trait StandardConfigFile:
         .map_err(SettingsError::ConfigFile)
         .map_err(Into::into)
     }
+
+    async fn layered_load_with_provider<F, P>(
+        app_name: impl AsRef<str> + Send,
+        parameter_provider: Option<(P, String)>,
+        set_func: F,
+    ) -> std::result::Result<Self, Self::Error>
+    where
+        F: FnOnce(&mut ConfigMap) + Send,
+        P: ParameterProvider + 'static,
+    {
+        let app_name = app_name.as_ref();
+        config_file::layered_load_with_provider(
+            app_name,
+            "toml",
+            &Some(format!("SI_{}_CONFIG", app_name.to_uppercase())),
+            &Some(format!("SI_{}", app_name.to_uppercase())),
+            parameter_provider,
+            set_func,
+        )
+        .await
+        .map_err(SettingsError::ConfigFile)
+        .map_err(Into::into)
+    }
+}
+
+#[allow(clippy::disallowed_methods)]
```

Comment:
Maybe there's a better place for this? I need to know a few things when innit-client does its work and I figured pulling out of env vars would be best. I suppose I could let innit-client run layered-load without the parameter bits to do this?

Reply:
I think this is the right call for now. I'd consider moving `local` to a const though with the `DEFAULT_` prefix